src/Lecture3.hs

```diff
@@ -156,10 +174,14 @@ monsters, you should get a combined treasure and not just the first
 🕯 HINT: You may need to add additional constraints to this instance
   declaration.
 -}
-instance Semigroup (Treasure a) where
+instance (Semigroup a) => Semigroup (Treasure a) where
+  (<>) a NoTreasure = a
+  (<>) NoTreasure b = b
+  (<>) (SomeTreasure a) (SomeTreasure b) = SomeTreasure $ a <> b
 
 
-instance Monoid (Treasure a) where
+instance (Monoid a) => Monoid (Treasure a) where
```

Comment:
The `Monoid` constraint is too strict here. You don't use `mempty` so you don't need `Monoid` and `Semigroup` is enough 👍🏻 
```suggestion
instance (Semigroup a) => Monoid (Treasure a) where
```